tool/tctl/common/user_command.go

```diff
@@ -111,35 +151,66 @@ func (u *UserCommand) Add(client auth.ClientI) error {
 	if u.kubeGroups != "" {
 		kubeGroups = strings.Split(u.kubeGroups, ",")
 	}
-	user := services.UserV1{
-		Name:          u.login,
-		AllowedLogins: strings.Split(u.allowedLogins, ","),
-		KubeGroups:    kubeGroups,
+
+	// make sure that user does not exist
+	_, err := client.GetUser(u.login, false)
+	if err == nil {
+		if err == nil {
+			return trace.BadParameter("user(%v) already registered", u.login)
+		}
 	}
-	token, err := client.CreateSignupToken(user, u.ttl)
+
+	user, err := services.NewUser(u.login)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	traits := map[string][]string{
+		teleport.TraitLogins:     strings.Split(u.allowedLogins, ","),
+		teleport.TraitKubeGroups: kubeGroups,
+	}
+
+	user.SetTraits(traits)
+	user.AddRole(teleport.AdminRoleName)
+	err = client.UpsertUser(user)
```

Comment:
Why isn't all of this logic within `CreateUserToken`?

Reply:
CreateUserToken is about resetting a password. In here,  `tctl users add xxx` creates a new user and generates a token to set up a password.